src/server/game/Spells/SpellMgr.cpp

```diff
@@ -4631,6 +4631,15 @@ void SpellMgr::LoadSpellInfoCorrections()
         spellInfo->AttributesEx &= ~SPELL_ATTR1_IS_CHANNELLED;
     });
 
+    // Hot Rod Radio
+    ApplySpellFix({ 66298 }, [](SpellInfo* spellInfo)
+    {
+        ApplySpellEffectFix(spellInfo, EFFECT_0, [](SpellEffectInfo* spellEffectInfo)
+        {
+            spellEffectInfo->TargetA = SpellImplicitTargetInfo(TARGET_UNIT_PASSENGER_0);
```

Comment:
hack.  TARGET_UNIT_TARGET_PASSENGER should target the first passenger by default. If that's not the case, the core needs a fixup.

Reply:
No, you manually target the passenger. That target allows any unit on the vehicle. So instead of hacking the spell to only target seat 0, just explicitely select the player.

Reply:
i just realize there is a typo error (is not the radio but the horn honk spell), and in the script i selected the player but still says no target without this hack

Reply:
So this is a bug then that needs to be fixed instead.